api/client/registry/logout.go

```diff
@@ -31,20 +32,43 @@ func NewLogoutCommand(dockerCli *client.DockerCli) *cobra.Command {
 
 func runLogout(dockerCli *client.DockerCli, serverAddress string) error {
 	ctx := context.Background()
+	var isDefaultRegistry bool
 
 	if serverAddress == "" {
 		serverAddress = dockerCli.ElectAuthServer(ctx)
+		isDefaultRegistry = true
+	}
+
+	var (
+		loggedIn        bool
+		regToLogout     string
+		hostnameAddress = serverAddress
+		regsToTry       = []string{serverAddress}
+	)
+	if !isDefaultRegistry {
+		hostnameAddress = registry.ConvertToHostname(serverAddress)
+		// the tries below are kept for backward compatibily where a user could have
+		// saved the registry in one of the following format.
+		regsToTry = append(regsToTry, hostnameAddress, "http://"+hostnameAddress, "https://"+hostnameAddress)
 	}
 
 	// check if we're logged in based on the records in the config file
 	// which means it couldn't have user/pass cause they may be in the creds store
-	if _, ok := dockerCli.ConfigFile().AuthConfigs[serverAddress]; !ok {
-		fmt.Fprintf(dockerCli.Out(), "Not logged in to %s\n", serverAddress)
+	for _, s := range regsToTry {
+		if _, ok := dockerCli.ConfigFile().AuthConfigs[s]; ok {
+			loggedIn = true
+			regToLogout = s
+			break
```

Comment:
Wondering if we should log out from all of the registries from `regsToTry` which are found, in case someone ends up in a state where the same registry is stored in multiple ways. It would be very unexpected if `docker logout` only removed one copy but not the others.